rust-src/wallet_library/src/statement.rs

```diff
@@ -0,0 +1,403 @@
+            "idDocIssuer should be allowed ISO3166-2 values"
+        );
+    }
+
+    #[test]
+    pub fn range_statements_are_only_allowed_on_some_tags_test() {
+        let dob_statement: AtomicStatement<
+            constants::ArCurve,
+            AttributeTag,
+            constants::AttributeKind,
+        > = AtomicStatement::AttributeInRange {
+            statement: AttributeInRangeStatement {
+                attribute_tag: AttributeTag(3),
+                lower:         constants::AttributeKind::from(20140101),
+                upper:         constants::AttributeKind::from(20150101),
+                _phantom:      PhantomData::<constants::ArCurve>,
+            },
+        };
+
+        let name_statement: AtomicStatement<
+            constants::ArCurve,
+            AttributeTag,
+            constants::AttributeKind,
+        > = AtomicStatement::AttributeInRange {
+            statement: AttributeInRangeStatement {
+                attribute_tag: AttributeTag(2),
+                lower:         constants::AttributeKind::from(20140101),
+                upper:         constants::AttributeKind::from(20150101),
+                _phantom:      PhantomData::<constants::ArCurve>,
+            },
+        };
+        assert!(
+            dob_statement
+                .acceptable_atomic_statement(get_default_wallet_config().identity_rules.as_ref())
+                .is_ok(),
+            "Range statement should be allowed on tag 3 (dob)"
+        );
+        assert!(matches!(
+            name_statement
+                .acceptable_atomic_statement(get_default_wallet_config().identity_rules.as_ref()),
+            Err(RequestCheckError::IllegalRangeTag(_))
+        ));
+    }
+
+    #[test]
+    pub fn multiple_statements_on_tag_test() -> anyhow::Result<()> {
+        let statement1: AtomicStatement<
+            constants::ArCurve,
+            AttributeTag,
+            constants::AttributeKind,
+        > = AtomicStatement::RevealAttribute {
+            statement: RevealAttributeStatement {
+                attribute_tag: AttributeTag(3),
+            },
+        };
+
+        let statement2: AtomicStatement<
+            constants::ArCurve,
+            AttributeTag,
+            constants::AttributeKind,
+        > = AtomicStatement::AttributeInRange {
+            statement: AttributeInRangeStatement {
+                attribute_tag: AttributeTag(3),
+                lower:         constants::AttributeKind::from(5),
+                upper:         constants::AttributeKind::from(10),
+                _phantom:      PhantomData::<constants::ArCurve>,
+            },
+        };
+
+        let statement: CredentialStatement<constants::ArCurve, constants::AttributeKind> = CredentialStatement::Account { network: Network::Testnet, cred_id: CredentialRegistrationID::from_str("8a3a87f3f38a7a507d1e85dc02a92b8bcaa859f5cf56accb3c1bc7c40e1789b4933875a38dd4c0646ca3e940a02c42d8")?, statement: vec![statement1, statement2]};
```

Comment:
That's a long line. Is there no formatting for this crate?

Reply:
When I run fmt, it doesn't care :thinking:
I thought it uses the rules that are specified in the root folder.
I don't think it is checked on the CI, so I'll add that in another PR.